test/Prototypes/Unicode.swift

```diff
@@ -6,6 +6,42 @@ import Swift
 import SwiftShims
 import StdlibUnittest
 
+protocol Unicode {
+  associatedtype Encoding: UnicodeEncoding
+  associatedtype CodeUnits: RandomAccessCollection
+  /* where CodeUnits.Iterator.Element == Encoding.CodeUnit */
+  var codeUnits: CodeUnits {get}
+
+  associatedtype ValidUTF8View : BidirectionalCollection
+  // where ValidUTF8View.Iterator.Element == UTF8.CodeUnit */
+  // = TranscodedView<CodeUnits, Encoding, UTF8>
+  var utf8: ValidUTF8View {get}
+
+  associatedtype ValidUTF16View : BidirectionalCollection
+  // where ValidUTF16View.Iterator.Element == UTF16.CodeUnit
```

Comment:
I have no particular preference. I'm sure there'll be enough other code to experiment with.